lib/cli/cli.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+
+const CommandProcessor = require('./command-processor');
+const LineProcessor  = require('./line-processor');
+const utils = require('./utils');
+
+module.exports = class Cli extends EventEmitter {
+  // log - logger function to print results
+  //       help and error messages
+  constructor(log) {
+    super();
+
+    this.log = log;
+
+    this.client = null;
+    this.connection = null;
+    this.api = {};
+
+    this.commandProcessor = new CommandProcessor(this);
+    this.commandProcessor.on('exit', () => this.emit('exit'));
+
+    this.lineProcessor = new LineProcessor(this.commandProcessor);
+
+    // Map all remaining commands directly
+    utils.filterKeys(
+      Object.getOwnPropertyNames(CommandProcessor.prototype),
+      ['constructor', 'complete', 'getNextCompleter']
+    ).map((command) => {
+      if (!this.lineProcessor[command]) {
+        this.lineProcessor[command] =
+          this.commandProcessor[command].bind(this.commandProcessor);
```

Comment:
One more thing: I'd find it more readable to be `(...args) => this.commandProcessor[command](...args)`, but I'm fine with it either way.